src/nxdoc/web-ui/web-ui-functional-testing.md

```diff
@@ -0,0 +1,163 @@
+---
+title: Web UI Functional Testing
+description: An overview of functional testing on Web UI and it's functional testing framework.
+toc: true
+review:
+  date: '2019-03-13'
+  status: ok
+  comment: ''
+---
+
+This page covers the bases of functional testing on Web UI and provides an overview of the Web UI functional tests framework.
+
+## Functional Testing Overview
+
+The Web UI functional tests are developed on top of [WebdriverIO](https://webdriver.io/) and [Cucumber](https://cucumber.io/), providing a mechanism to run webdriver tests written in plain language (English). The advantages of this approach are threefold: 1) it improves readability, 2) it makes it easier to write tests based on requirements, specifications, and acceptance criteria, and ultimately, 3) it facilitates test writing by people other than developers.
+
+Cucumber is a crucial element of our testing stack, promoting Behavior-Driven Development (or **BDD** for short). Cucumber tests are written in [Gherkin](https://docs.cucumber.io/gherkin/reference/), which are then translated into Javascript instructions that run commands using the webdriver protocol. Tests are written in **feature** files (ending with the *.feature* extension), which are composed by a set of **Scenarios** with one ore more **Steps**. Steps common to more than one scenario are usually placed in a **Background**, which will be run before every scenario. Steps must begin with one of the following keywords: *Given*, *When*, *Then*, *And*, *But*. Which keyword is used only affects it's meaning to readers, although the *Given* keyword is commonly used for provisioning steps.
+
+{{#> panel type='code' heading='Sample feature file'}}
+
+```Gherkin
+Feature: Login / Logout
+
+  As an user I can login and logout
+
+  Scenario: Login as member user
+    Given user "John" exists in group "members"
+    When I login as "John"
+    Then I am logged in as "John"
+```
+
+{{/panel}}
+
+
+{{#> callout type='note'}}
+Please check the [Gherkin reference](https://docs.cucumber.io/gherkin/reference/) page for more details on how to write Cucumber tests.
+{{/callout}}
+
+Steps used in feature files are mapped into javascript instructions, which are read from **step definition** files. These are Javascript files that map a given set of steps into javascript code. These definitions usually rely on page helpers to access interface elements and assert the conditions matching the test criteria. **Page helpers** are objects that encapsulate the logics required to access and manipulate high level interface elements, hiding most of the webdriver-specific logics.
```

Comment:
Should pick one for consistency throughout the doc: Javascript, javascript or JavaScript?